src/menusys.c

```
@@ -91,6 +91,45 @@ static submenu_list_t *appMenu;
+ *
+ * This one is held only for the list work itself -- a few frees and mallocs, or one traversal --
+ * so the worst a waiter ever sees is microseconds.
+ *
+ * CALLER-LOCKED. menuAddHint()/menuRemoveHints() do NOT lock internally: a rebuild is many calls and
+ * must be atomic as a whole, and this semaphore is not recursive. Every caller brackets its own
+ * sequence with menuHintsLock()/menuHintsUnlock().
+ */
+static s32 menuHintSemaId = -1;
+static ee_sema_t menuHintSema;
+
+void menuHintsLock(void)
```

Cppcheck (2.21.0)

[style] The function 'menuHintsLock' should have static linkage since it is not used outside of its translation unit.

(staticFunction)